bgp: extract exportPodCIDRReconciler logic into a generic function

jibi opened the pull request:
This commit extracts most of the exportPodCIDRReconciler logic into a
more generic function, exportAdvertisementsReconciler, so that it can be
reused by other subsystems which only need to advertise a set of IPNets.

jibi committed:
bgp: move exportPodCIDRReconciler to advertisements_reconciler.go

No functional changes, but this should make it easier to review the next
commit.

Signed-off-by: Gilberto Bertin <jibi@cilium.io>

lmb commented on pkg/bgpv1/gobgp/advertisements_reconciler.go:
```diff
@@ -112,48 +110,47 @@ func exportPodCIDRReconciler(ctx context.Context, _ *BGPRouterManager, sc *Serve
 	if len(toAdvertise) == 0 && len(toWithdraw) == 0 {
 		l.Debugf("No reconciliation necessary")
-		return nil
+		return params.currentAdvertisements, nil
```
Should this return a copy of the slice?

jibi committed:
bgp: extract exportPodCIDRReconciler logic into a generic function

This commit extracts most of the exportPodCIDRReconciler logic into a
more generic function, exportAdvertisementsReconciler, so that it can be
reused by other subsystems which only need to advertise a set of IPNets.

Signed-off-by: Gilberto Bertin <jibi@cilium.io>
